docs/monorepo.md

```diff
@@ -0,0 +1,79 @@
+
+All GitHub Actions configuration for the monorepo, including CI, lives in `.github`. We should strive to make things here generic rather than specifc to any one project.
+
+* Actual actions live in `.github/actions/`. If it doesn't have a `action.yml` file, it shouldn't be in there.
+* Pattern matchers (not associated with an action), go in `.github/matchers/`.
+* Other files specific to actions, including scripts used with `run:`, go in `.github/files/`.
+
+### Compatibility
+
+All projects should be compatible with PHP versions WordPress supports. That's currently PHP 5.6 to 8.0.
+
+We use eslint and phpcs to lint JavaScript and PHP code. Projects should comply with the standards enforced by these tools.
+
+* Projects may include `.eslintrc.js` to adjust eslint configuration as necessary, but try to keep to the spirit of it.
+* As eslint does not support per-directory `.eslintignore`, any necessary ignore rules should be added to the file in the monorepo.
+* As phpcs does not support per-directory configuration, any necessary file ignore clauses should be added to `.phpcs.config.xml`.
+
+### Project structure
+
+We use `composer.json` to hold metadata about projects. Much of out generic tooling reads this metadata to customize handling of the project. Metadata keys used are:
+
+* `.name`: Generally "Automattic/jetpack-_something_". Used to report names in various places. For Composer packages, this must, of course, match the name on Packagist.
+* `.version`: If present, updated by `tools/plugin-version.sh`.
+* `.repositories`: If you include a repository entry referencing monorepo packages, it must have `.options.monorepo` set to true. This allows the build tooling to recognize and remove it.
+* `.scripts.build-production`: If your project requires a build step, this must run the necessary commands.
+* `.scripts.phpunit`: Packages must either set this to run PHPUnit tests, or must include a file `tests/php/ci-can-run.sh` that exits with a failure status.
+* `.extra.mirror-repo`: This specifies the name of the GitHub mirror repo, i.e. the "Automattic/jetpack-_something_" in "https://github.com/Automattic/jetpack-_something_".
+* `.extra.release-branch-prefix`: Our mirroring and release tooling considers any branch named like "_prefix_/branch-_version_" to be a release branch, and this specifies which _prefix_ belongs to the project.
+* `.extra.version-constants`: When `tools/plugin-version.sh` is updating versions, this specifies PHP constants to replace. The value is an object matching constants to the file (relative to the plugin root) in which the constant is defined.
+  * Note that constant definitions must be on a single line and use single quotes to be detected by the script. Like this:
+    ```php
+    define( 'CONSTANT', 'version' );
+    ```
+* `.extra.wp-plugin-name`: Specify the WordPress.org plugin name, for use by scripts that deploy the plugin to WordPress.org.
```

Comment:
```suggestion
* `.extra.wp-plugin-name`: Specify the WordPress.org plugin slug, for use by scripts that deploy the plugin to WordPress.org.
```
Suggesting to be more exact that we need their slug (e.g. wordpress.org/plugins/$SLUG)